### PR TITLE
checkout step 1 validation

### PR DESCRIPTION
**Pull request by adammeyer**

### CODE REVIEW

File: src/app/checkout/step-1/step-1.component.js [hunk 1]
```diff
@@ -67,6 +69,6 @@ export default angular
     controller: Step1Controller,
     templateUrl: template.name,
     bindings: {
-      changeStep: '&'
```
Comment (adammeyer) [inline src/app/checkout/step-1/step-1.component.js]: @OzzieOrca I wasn't able to get the function to change the step, so I changed it to bind the variable.  Any downsides to this?

**Comment by adammeyer** [comment]
@OzzieOrca I'm merging this now, but let me know if you have any changes and I'll correct them.  thanks!

File: src/app/checkout/step-1/step-1.component.js (hunk 1 above)
Comment (OzzieOrca) [inline src/app/checkout/step-1/step-1.component.js]: It adds watchers... Not a big deal. I think you need to call it like [this](https://github.com/CruGlobal/give-web/pull/6/files#diff-723383532014b6b0bdf61e792b226b2cR64). 

So you'd use it like this:
```
<my-component change-step="$ctrl.parentComponentFunction(param1)"></my-component>
```
Parent component controller:
```
parentComponentFunction(someParam1){
  //do something
}
```
Child component:
```
this.changeStep({param1: 'value})
```

For some reason it uses named parameters so the key in the child component has to match what you put in the template. I guess it has to parse the function in the template as a string and then call it.

Comment (adammeyer) [inline src/app/checkout/step-1/step-1.component.js]: I gave that a try this morning, but I couldn't figure it out.  I think there must be something missing.

**Pull request merged**